src/main/groovy/com/bmuschko/gradle/docker/tasks/network/DockerCreateNetwork.groovy

```diff
@@ -25,7 +32,14 @@ class DockerCreateNetwork extends AbstractDockerRemoteApiTask {
 
     void runRemoteCommand() {
         logger.quiet "Creating network '${networkName.get()}'."
-        CreateNetworkResponse network = dockerClient.createNetworkCmd().withName(networkName.get()).exec()
+
+        CreateNetworkCmd networkCmd = dockerClient.createNetworkCmd().withName(networkName.get())
+
+        if (subnet.getOrNull()) {
+            networkCmd.withIpam(new Network.Ipam().withConfig(new Network.Ipam.Config().withSubnet(subnet.get())))
```

Comment:
We have to model this functionality thinking forward into the future. We should introduce an inner class `Ipam` and then have a list of `Config` similar to what Docker Java does. Instead of a field `subnet`, we'd have a field called `ipam`. With that we'll be able to easily add more fields going forward.

Reply:
Pushed a commit to add the inner Ipam class which expects a list of maps to parse (similar to how [docker-java's network ipam](https://github.com/docker-java/docker-java/blob/c3661f3da85ebf4f9caea5a75d1c9224d2937876/docker-java-api/src/main/java/com/github/dockerjava/api/model/Network.java#L138) has a list of Configs). I had wanted to use `@Nested` within `@Nested` so we would have 
```
@Nested
@Optional
ListProperty<Config> config
```

Instead of the generic map, but had trouble getting that to work. Please let me know if you have any suggestions, I'm very new to both groovy and the Gradle API.

Reply:
Hhmmm, did you try to create that with `objectFactory.newInstance` as well? Maybe that's a Gradle limitation. What exactly didn't work?